### PR TITLE
colorize nestLike format

### PR DESCRIPTION
**Pull request by mizozobu**
Colorize nestLike format based on log level. Same colors used in `@nest/common` logger.

### CODE REVIEW

**Comment by gremo** [comment]
Hi! Thanks for contributing. Source of the color scheme from NestJS?

**Comment by mizozobu** [comment]
Right here.
https://github.com/nestjs/nest/blob/master/packages/common/services/logger.service.ts#L73-L97

**Comment by gremo** [comment]
I see, Ok. I've added a few annotations in your PR. Another question: wouldn't be better to use `cli-color` (like the nest logger) instead of `colors`, as library to manage colors? That way, color match would be extract. What do you think?

**Comment by mizozobu** [comment]
I can see 2 benefits of switching to `cli-color`.

1. Reduce the bundle size of projects using `nest-winston` by removing `colors` from its dependency.
2. Match the exact color scheme of NestJS logger.

File: src/winston.utilities.ts [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+  error: clc.red,
+  warn: clc.yellow,
+  debug: clc.magentaBright,
+  verbose: clc.cyanBright,
+};
+
 const nestLikeConsoleFormat = (appName = 'NestWinston'): Format => format.printf(({ context, level, timestamp, message, ...meta }) => {
-  return `${green(`[${appName}]`)} ` +
-         `${yellow(level.charAt(0).toUpperCase() + level.slice(1))}\t` +
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (mizozobu) [inline src/winston.utilities.ts:16]: Without this, eslint warns `Unnecessary conditional, value is always truthy.` Is it acceptable?

Comment (gremo) [reply src/winston.utilities.ts:16]: Yes, very good solution!

File: src/winston.utilities.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+  const color = nestLikeColorScheme[level] || ((text: string): string => text);
```
Comment (gremo) [inline src/winston.utilities.ts:17]: Just a personal taste, can you rename to `const colorize`? It sounds more like a function.

**Comment by gremo** [comment]
I've just added a trivial change, then I will merge.

I've one doubt. Since this package can be used with Nest 5.x 6.x and 7.x, what about people still using Nest 5.x or 6.x where (maybe, I don't now) the color scheme wasn't introduced? I mean, they will get colorized output from this module but uncolorized output from Nest js, right?

**Comment by mizozobu** [comment]
I started using NestJS after v7, so I never knew what it was like back then. If I'm not mistaken, NestJS logger had colorized output even back in v5.0.0.
https://github.com/nestjs/nest/blob/v5.0.0/packages/common/services/logger.service.ts#L57-L73

`nestLike` format has been outputting logs in green. Maybe it's just me but going from green to more colors seems more natural than going to no colors.

**Comment by gremo** [comment]
> I started using NestJS after v7, so I never knew what it was like back then. If I'm not mistaken, NestJS logger had colorized output even back in v5.0.0.
> https://github.com/nestjs/nest/blob/v5.0.0/packages/common/services/logger.service.ts#L57-L73
> 
> `nestLike` format has been outputting logs in green. Maybe it's just me but going from green to more colors seems more natural than going to no colors.

Perfect, didn't remember that. Let's merge and release, and.. thanks for helping!

**Pull request merged**